simplex-chat/simplex-chat
# android: Fallback to manual parsing of apiChats and apiChat responses

To test it, add
```kotlin
val str = str.replace("itemText\":", "asd\":")
```
before this line: https://github.com/simplex-chat/simplex-chat/blob/b9d5b2ad7e44c8ab7b39e0a82cece04d9c9d97a8/apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt#L2579

After that it should work without problems too.

Then remove 
```kotlin
decodeObject(ChatInfo.serializer(), chat.jsonObject["chatInfo"])
    ?: 
```

on line https://github.com/simplex-chat/simplex-chat/blob/b9d5b2ad7e44c8ab7b39e0a82cece04d9c9d97a8/apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt#L2619

Main chat list will show invalid items

Then return back that line and remove this:
```kotlin
chat.jsonObject["chatItems"]!!.jsonArray.map {
    decodeObject(ChatItem.serializer(), it) ?:
```
here: https://github.com/simplex-chat/simplex-chat/blob/b9d5b2ad7e44c8ab7b39e0a82cece04d9c9d97a8/apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt#L2622

Chat list will be fine while chat items will be invalid.
